src/libos/src/unit_test/proc_macro/Cargo.toml

```diff
@@ -0,0 +1,13 @@
+[package]
+name = "occlum_test_proc_macro"
+version = "0.0.1"
+edition = "2018"
+
+[lib]
+proc-macro = true
+
+[dependencies]
+proc-macro2 = "1.0.24"
+quote = "1.0.7"
+syn = { version = "1.0.42", features = ["parsing"] }
+inventory = "0.1.9"
```

Comment:
Having a separate crate under `libos/src` breaks the current unwritten convention of code organization. Having a crate embedded in the directory of another crate does not seem to be a good practice for Rust projects. The best solution that comes to my mind right now is to add a `crates` directory under `libos` directory. This approach is clear and extensible: more crates can be added in the future.

Could you try rewriting the `unit_test` module and the `proc_macro` crate as separate crates under the `crates` directory? Also, can we combine them into one crate?

Reply:
I added `proc_macro` in the crates directory. It has no dependencies on occlum. Instead, `unit_test` module relies on occlum heavily. I leave the modularization for future work.